python/cudf/cudf/core/frame.py

```diff
@@ -2489,9 +2492,14 @@ def _colwise_binop(
                         left_column = left_column.fillna(fill_value)
                     elif right_column.nullable:
                         right_column = right_column.fillna(fill_value)
-                else:
+                elif left_is_column:
                     if left_column.nullable:
                         left_column = left_column.fillna(fill_value)
+                elif right_is_column:
+                    if right_column.nullable:
+                        right_column = right_column.fillna(fill_value)
+                else:
+                    assert False, "At least one operand must be a column."
```

Comment:
```suggestion
                    raise TypeError("At least one operand must be a column.")
```

Should we convert this into a `TypeError`?

Reply:
No. This will only result from developer error, not user error. This function is internal and is called relatively deep in the binary operations logic, so this error should only arise if a developer does something wrong.